components/producers/producer.go

```diff
@@ -160,3 +162,21 @@ func getSource() string {
 	}
 	return strings.TrimSpace(string(dat))
 }
+
+// GetPURLTarget returns a purl target string for a given package.
+// This should be used as the `Issue.Target` field of SCA producers.
+//
+// Example: GetPURLTarget("deb", "debian", "curl", "7.68.0", nil, "")
+func GetPURLTarget(purlType string, namespace string, name string, version string, qualifiers packageurl.Qualifiers, subpath string) string {
+	return packageurl.NewPackageURL(purlType, namespace, name, version, qualifiers, subpath).ToString()
+}
+
+// EnsureValidPURLTarget takes a purl target string from an untrusted source, e.g. a tool output, and
+// ensures it is a valid purl target according to the packageurl-go library.
+func EnsureValidPURLTarget(purlTarget string) (string, error) {
+	instance, err := packageurl.FromString(purlTarget)
+	if err != nil {
+		return "", err
+	}
+	return instance.ToString(), nil
```

Comment:
```suggestion
	return purlTarget, nil
```

Reply:
mmh, no, I'd rather use the string where we validated it's correct, instead of the input. The implementation of `FromString` does a couple modifications on the input, so it's possible that the input changes